codegen/module.go

```diff
@@ -1022,6 +1022,134 @@ func (system *ModuleSystem) getSpec(instance *ModuleInstance) (interface{}, bool
 	return spec, true, nil
 }
 
+func (system *ModuleSystem) tryResolveIncrementalBuild(
+	instances []ModuleDependency,
+	resolvedModules map[string][]*ModuleInstance,
+) (map[string][]*ModuleInstance, error) {
+
+	toBeBuiltModules := make(map[string]map[string]*ModuleInstance, 0)
```

Comment:
nit: rename `toBeBuiltModules` to `toBeBuiltModulesUnique` to note why it's a `map[string]map[string]*ModuleInstance`.